### PR TITLE
tests: Add unit test for libmptcpwrap.so.

### PR DESCRIPTION
**Pull request by ossama-othman**

### CODE REVIEW

File: tests/test-mptcpwrap [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#! /bin/sh
+# SPDX-License-Identifier: BSD-3-Clause
+
+# Test that libmptcpwrap injects MPTCP into the socket() system call.
+#
+# Copyright (c) 2021, Intel Corporation
+
+set -e
+
+LD_PRELOAD=../src/.libs/libmptcpwrap.so ./mptcpwrap-tester
```
Comment (mjmartineau) [inline tests/test-mptcpwrap:10]: Do you think it's better to use `LD_PRELOAD` or to use a "normal" dynamic link?

**Review by mjmartineau — APPROVED** [review]
I think it's fine to merge as-is, one question about linking

File: tests/test-mptcpwrap (hunk 1 above)
Comment (ossama-othman) [reply tests/test-mptcpwrap:10]: I was under the impression that the general use case was to preload libmptcpwrap.so so I did that for the test as well,, but I'm fine with "normal" dynamic link if you prefer.

Comment (mjmartineau) [reply tests/test-mptcpwrap:10]: Given that our "normal" use case is LD_PRELOAD, probably best to stay close to that

**Pull request merged**